pkg/cmdline/cmd_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cmdline
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/spf13/cobra"
+)
+
+var parentCmd = &cobra.Command{Use: "parent"}
+var childCmd = &cobra.Command{Use: "child"}
+
+func newCommandManager(cmd *cobra.Command) (cm *CommandManager, err error) {
+	defer func() {
+		if t := recover(); t != nil {
+			err = fmt.Errorf("%s", t)
+		}
+	}()
+	return NewCommandManager(cmd), nil
+}
+
+func registerCmd(cm *CommandManager, cmd *cobra.Command) (err error) {
+	defer func() {
+		if t := recover(); t != nil {
+			err = fmt.Errorf("%s", t)
+		}
+	}()
+	cm.RegisterCmd(cmd, false)
+	return
+}
+
+func registerSubCmd(cm *CommandManager, parent, child *cobra.Command) (err error) {
+	defer func() {
+		if t := recover(); t != nil {
+			err = fmt.Errorf("%s", t)
+		}
+	}()
+	cm.RegisterSubCmd(parent, child, false)
+	return
+}
+
+func setCmdGroup(cm *CommandManager, name string, cmds ...*cobra.Command) (err error) {
+	defer func() {
+		if t := recover(); t != nil {
+			err = fmt.Errorf("%s", t)
+		}
+	}()
+	cm.SetCmdGroup(name, cmds...)
+	return
+}
+
+func TestCommandManager(t *testing.T) {
+	_, err := newCommandManager(nil)
+	if err == nil {
+		t.Errorf("unexpected success with root nil command")
+	}
+	cm, err := newCommandManager(parentCmd)
+	if err != nil {
+		t.Errorf("unexpected error while instantiating new command manager: %err", err)
+	}
+	if cm.GetRootCmd() != parentCmd {
+		t.Errorf("unexpected root command returned")
+	}
+}
+
+func TestRegisterCmd(t *testing.T) {
+	cm, err := newCommandManager(parentCmd)
+	if err != nil {
+		t.Errorf("unexpected error while instantiating new command manager: %err", err)
+	}
+
+	if err := registerCmd(cm, nil); err == nil {
+		t.Errorf("unexpected success with nil command")
+	}
+
+	if err := registerCmd(cm, childCmd); err != nil {
+		t.Errorf("unexpected error while registering command: %s", err)
+	}
+
+	if cm.GetCmd("nochild") != nil {
```

Comment:
where do these strings come from?

Reply:
This is the check for non-existent command. Will add comment